tests/headers/referer_test.go

```diff
@@ -105,3 +93,57 @@ func TestReferer(t *testing.T) {
 		})
 	}
 }
+
+func TestRefererOrdering(t *testing.T) {
+	// It is not clear from the documentation of http.Request.Referer()
+	// that only the first Referer header is used and any other ones
+	// are ignored. This could potentially lead to security issues if two
+	// HTTP servers are chained that look at different headers.
+	//
```

Comment:
The documentation of http.Request.Referer() doesn't clearly specify whether only the first Referer header is used and the other ones ignored. This could potentially lead to security issues if two HTTP servers that look at different headers are chained together.